sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java

```diff
@@ -0,0 +1,154 @@
+package com.microsoft.appcenter.analytics;
+
+import android.support.annotation.NonNull;
+
+import com.microsoft.appcenter.channel.AbstractChannelListener;
+import com.microsoft.appcenter.channel.Channel;
+import com.microsoft.appcenter.ingestion.models.Log;
+import com.microsoft.appcenter.ingestion.models.one.AppExtension;
+import com.microsoft.appcenter.ingestion.models.one.CommonSchemaLog;
+
+/**
+ * Allow overriding Part A properties.
+ */
+public class PropertyConfigurator extends AbstractChannelListener {
+
+    /**
+     * App name to override common schema part A 'app.name'.
+     */
+    private String mAppName;
+
+    /**
+     * App name to override common schema part A 'app.ver'.
+     */
+    private String mAppVersion;
+
+    /**
+     * App name to override common schema part A 'app.locale'.
+     */
+    private String mAppLocale;
+
+    /**
+     * The transmission target which this configurator belongs to.
+     */
+    private AnalyticsTransmissionTarget mTransmissionTarget;
+
+    /**
+     * Create a new property configurator.
+     *
+     * @param channel            The channel for this listener.
+     * @param transmissionTarget The tranmission target of the configurator.
+     */
+    PropertyConfigurator(Channel channel, AnalyticsTransmissionTarget transmissionTarget) {
+        mTransmissionTarget = transmissionTarget;
+        if (channel != null) {
+            channel.addListener(this);
+        }
+    }
+
+    /**
+     * Override or inherit common schema properties while preparing log.
+     *
+     * @param log       A log.
+     * @param groupName The group name.
+     */
+    @Override
+    public void onPreparingLog(@NonNull Log log, @NonNull String groupName) {
+        if (log instanceof CommonSchemaLog && mTransmissionTarget.isEnabled()) {
+            AppExtension app = ((CommonSchemaLog) log).getExt().getApp();
+
+            /* Override app name if not null, else use the name of the nearest parent. */
+            if (mAppName != null) {
+                app.setId(mAppName);
+            } else {
+                for (AnalyticsTransmissionTarget target = mTransmissionTarget.mParentTarget; target != null; target = target.mParentTarget) {
+                    String parentAppName = target.getPropertyConfigurator().getAppName();
+                    if (parentAppName != null) {
+                        app.setId(parentAppName);
```

Comment:
This is `app.id`, not `app.name`.

Reply:
You mean the opposite? It should call `setName`

Reply:
Yeah I added those

Reply:
I meant the implementation is overriding `app.id`, not `app.name` and it should be `app.name`.
Sorry for confusion but I think we are on the same page. 😄